transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/SchedulerIntegrationTest.java

```diff
@@ -0,0 +1,185 @@
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import com.google.android.datatransport.Event;
+import com.google.android.datatransport.Transport;
+import com.google.android.datatransport.TransportFactory;
+import com.google.android.datatransport.runtime.backends.BackendRegistry;
+import com.google.android.datatransport.runtime.backends.TransportBackend;
+import com.google.android.datatransport.runtime.scheduling.jobscheduling.SchedulerConfig;
+import com.google.android.datatransport.runtime.scheduling.jobscheduling.Uploader;
+import java.nio.charset.Charset;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+@RunWith(AndroidJUnit4.class)
+public class SchedulerIntegrationTest {
+  private static final String TEST_KEY = "test";
+  private static final String TEST_VALUE = "test-value";
+  private static final String testTransport = "testTransport";
+  private TransportInternal transportInternalMock = mock(TransportInternal.class);
+  private TransportBackend mockBackend = mock(TransportBackend.class);
+  private TransportBackend mockBackend2 = mock(TransportBackend.class);
+  private BackendRegistry mockRegistry = mock(BackendRegistry.class);
+  private final Context context = InstrumentationRegistry.getInstrumentation().getContext();
+  private final Uploader mockUploader = mock(Uploader.class);
+  private TransportRuntime runtime;
+
+  @Before
+  public void setUp() {
+    TransportRuntime.setInstance(
+        DaggerTestRuntimeComponent.builder()
+            .setApplicationContext(context)
+            .setUploader(mockUploader)
+            .setBackendRegistry(mockRegistry)
+            .setSchedulerConfig(new SchedulerConfig(500, 100000, 500))
+            .setEventClock(() -> 3)
+            .setUptimeClock(() -> 1)
+            .build());
+    when(mockBackend.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    when(mockBackend2.decorate(any()))
+        .thenAnswer(
+            (Answer<EventInternal>)
+                invocation ->
+                    invocation
+                        .<EventInternal>getArgument(0)
+                        .toBuilder()
+                        .addMetadata(TEST_KEY, TEST_VALUE)
+                        .build());
+    runtime = TransportRuntime.getInstance();
+  }
+
+  String getBackendName() {
+    byte[] array = new byte[8]; // length is bounded by 8
+    new Random().nextBytes(array);
+    return new String(array, Charset.forName("UTF-8"));
+  }
+
+  @Test
+  public void firstAttemptSchedule() {
```

Comment:
The test naming pattern elsewhere in the codebase is `methodOrSystemUnderTest_whenConditions_shouldDoStuff`. Could you rename them to the same pattern pls? e.g. `scheduler_whenEventScheduledForFirstTime_shouldUploadEventWithinExpectedTime` or something